### DISCUSSION TITLE
2.9.0-preview

### DISCUSSION THREAD

**Discussion by arianneorpilla** [arianneorpilla/jidoujisho#309] [Announcements]
This is a preview release for  [**2.9**](https://github.com/lrorpilla/jidoujisho/releases/tag/2.8.0). Please refer to the [**releases page**](https://github.com/lrorpilla/jidoujisho/releases) for a full timeline of changes. 

**System Requirements**: This application has been tested to work on a variety of devices and emulators running Android 7 to 13. Please note that a significant amount of internal storage space may be taken up by the dictionary database for indexing in order to optimize query speed and performance. Try to give the application around 2-4GB of storage space. Make sure your Android WebView is updated before use. This will resolve common issues that may occur with ッツ Ebook Reader or Mokuro.

# 2.9.0
- **Yomichan Structured Content Support** - The app now supports importing structured content Yomichan dictionaries, with images, tables and other structured content. Particularly, the app has been tested to properly import [**Aquafina's JMdict releases**](https://github.com/Aquafina-water-bottle/jmdict-english-yomichan/) and the [**Yomichan project's test dictionary**](https://github.com/FooSoft/yomichan/tree/master/test/data/dictionaries/valid-dictionary1).
- **Dictionary Import Improvements** - Imports should be significantly faster and use less memory. Lower end devices that experienced slowdowns and halts when importing larger dictionaries should see significant improvements.
- **Android 14 Fixes** - Fixes #302 where a black screen would occur when the user leaves the app in the background when a WebView is displayed.

### Known Issues and Notes
- Users may have to clear their dictionary database and re-import their dictionaries.
- The structured content is rendered differently and uses Flutter's `SelectionArea` instead of `SelectableText`. Double tap to smart select does not work for recursive searches. Users are advised to use drag to select at the moment.

# 📲 Which file should I download?
As a rule of thumb, `arm64` should work for most modern mobile devices. `armeabi` should work on older devices. Generally, desktops and emulators should use `x86_64`.

Depending on which device you are using, download the appropriate `APK` by expanding the `Assets` drop-down below and install on your mobile phone by launching the `APK` with your file explorer.

<hr /><em>This discussion was created from the release <a href='https://github.com/lrorpilla/jidoujisho/releases/tag/'>2.9.0-preview</a>.</em>

**Comment by Lyhcan** [comment]
I imported some of the English dictionaries from the Aquafina's link you provided, but when i search for a word it doesn't find anything it says "No search result found." Am i missing something? 

Edit: Switching the app's language did the trick, it seems the english dictionaries are not en-en as I thought, but jp-en or something, which is unfortunate.

wb btw :)

**Comment by arekito** [comment]
De verdad muchas gracias por continuar este proyecto, lo valoro mucho y me ha ayudado un montón con la inmersión del idioma japonés y ahora con la nueva función para inglés es mucho mejor!
Cómo podría donar a este proyecto de verdad tiene mucho potencial!

**Comment by epistularum** [comment]
I'm excited to see that work is being done to support structured content as new dictionaries tend to make use of it. However it seems pretty broken, see images below with reference picture taken from yomichan.
![image](https://github.com/lrorpilla/jidoujisho/assets/34507493/c4000ab6-44d0-4471-a2fe-843722c0ef4c)

![image](https://github.com/lrorpilla/jidoujisho/assets/34507493/e2f9c3d7-c998-41d7-8831-fc6efddb09d7)
